### PR TITLE
[bluetooth.bluegiga] Introduced better initialization functionality

### PR DESCRIPTION
**Pull request by paulianttila**
+ other improvements

Signed-off-by: Pauli Anttila <pauli.anttila@gmail.com>

### CODE REVIEW

**Comment by paulianttila** [comment]
Bluegiga error situations like framing errors happens very rarely, I tried to generate them by hand and with these modifications, bluegiga binding seems to be much more reliable than before.

File: bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java [hunk 1]
```diff
@@ -152,12 +153,14 @@
     private final Map<Integer, BluetoothAddress> connections = new ConcurrentHashMap<>();
 
     private volatile boolean initComplete = false;
+    private volatile boolean initRequest = false;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java:156]: Is this boolean even necessary? Would just checking `serialHandler.get().isAlive()` suffice?

File: bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java [hunk 2]
```diff
@@ -168,11 +171,40 @@ public BlueGigaBridgeHandler(Bridge bridge, SerialPortManager serialPortManager)
...
+                }
+            } catch (InterruptedException e) {
+                return;
+            } catch (ExecutionException e) {
+                initRequest = true;
+            }
+        }
+
+        if (initRequest) {
+            logger.debug("Initiliaze BlueGiga");
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java:198]:
```suggestion
            logger.debug("Initialize BlueGiga");
```

Comment (paulianttila) [reply bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java]: Done

File: bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java (hunk 1 above)
Comment (paulianttila) [reply bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java]: Yes and done

**Review by cpmeister — APPROVED** [review]

**Pull request merged**